### ISSUE TITLE
Support for du command

### ISSUE THREAD

**Issue by tlightsky** [nushell/nushell#679]
like the `du` command in linux or osx

**Comment by jonathandturner** [comment]
We don't have aliases yet to make it short, but you can get `du`-like data from:

```
sys | get disks
```

**Comment by tlightsky** [comment]
```
----------+----------+----------+------------+----------
 host     | cpu      | mem      | net        | battery 
----------+----------+----------+------------+----------
 [object] | [object] | [object] | [20 items] | [1 item] 
----------+----------+----------+------------+----------
```
my sys seems different

**Comment by jonathandturner** [comment]
@tlightsky - what system and version of the system are you on? It looks like the software we use to detect the disks isn't working

**Comment by tlightsky** [comment]
OSX 10.14.6
nushell 0.1.3

**Comment by jonathandturner** [comment]
@tlightsky - are you able to build nushell from source? We've done fixes to the `sys` command since that release, which might help fix the issue you're seeing.  We'll also be putting out a release in a few weeks, which will also have the fixes.

**Comment by tlightsky** [comment]
`cargo run` in nushell master branch
```
> sys 
error: No such file or directory (os error 2)
```

**Comment by jonathandturner** [comment]
@tlightsky - when building you may need to do `cargo build && cargo run` to make sure all the plugins are built.

**Comment by tlightsky** [comment]
still no disks
```
/Users/me/source/rust/nushell(master)> sys
━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━
 host           │ cpu            │ mem            │ net              │ battery 
────────────────┼────────────────┼────────────────┼──────────────────┼────────────────
 [table: 1 row] │ [table: 1 row] │ [table: 1 row] │ [table: 20 rows] │ [table: 1 row] 
━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━
```
btw, what i want is the `du` command to list all folders and the size of them inside current folder,
not `df` command which give me an overview of sytem disks status

**Comment by andrasio** [comment]
I can confirm `disks` isn't displayed on my mac.

**Comment by jonathandturner** [comment]
@svartalf - do you happen to know why the disks aren't showing up for some folks?

**Comment by jonathandturner** [comment]
@svartalf - as long as the version is published on crates.io, I'm okay moving to it

**Comment by thegedge** [comment]
Verified that `sys | get disks` is working for me with `0.8.0-alpha.1` (OSX 10.14.6). `0.8.0` is released now though, so I could go ahead and bump.

**Comment by jonathandturner** [comment]
@thegedge sounds good, if you want to send a PR I'll approve

**Comment by thegedge** [comment]
Looks like https://github.com/nushell/nushell/pull/813 actually bumped it in `Cargo.lock`. Probably no need to worry about bumping until the next release?

**Comment by jonathandturner** [comment]
Ah, that's true. I wonder - still might be good to bump in the Cargo.toml too, so everyone gets the same version. I forgot to do that.

**Comment by thegedge** [comment]
Sounds good! I'll send a quick PR.

**Comment by jonathandturner** [comment]
I'm going to go ahead and close since the original issue should be fixed. I'd like to be able to add aliases to be able to squeeze this down to a `du` alias, but we're tracking that elsewhere.

Thanks again.

**Comment by thegedge** [comment]
I intended to mention that one thing `du` does that `sys` doesn't seem to be able to provide is usage for directories. For example, in the `nushell` source directory:

```
$ du -hd1
 32K	./docker
 36K	./debian
2.9M	./images
3.2G	./target
172K	./tests
148K	./docs
  0B	./.cargo
8.0K	./.github
8.4M	./.git
524K	./assets
8.0K	./.circleci
4.0K	./.azure
1.2M	./src
3.3G	.
```

`sys | get disks` seems more akin to `df`.

**Comment by jonathandturner** [comment]
That's a good point. I was reading one and thinking the other. Let me reopen in case someone wants to create the du equivalent.

**Comment by andrasio** [comment]
#900

**Comment by Freaky** [comment]
A `du` command was merged in d48f99c - albeit a naive one.  Compare/contrast:

```
 12 │ target    │   1.9 GB │ [table 2 rows]

833M    ./target # du --si
1.6G    ./target # du --si -A
1.9G    ./target # du --si -Al
```

i.e. it considers only the apparent size of files, and doesn't do hardlink tracking for files with link counts >1.

**Comment by coolshaurya** [comment]
I believe this issue was fixed in version 0.9.0

**Comment by thegedge** [comment]
Agree with @Freaky that there are improvements we can make, but @coolshaurya is right that this was already resolved by #1250.

For specific improvements to `du`, we can open additional issues 🙂

Thanks everyone!
